source/contracts/reporting/AuctionToken.sol

```diff
@@ -0,0 +1,85 @@
+pragma solidity 0.4.24;
+
+import 'reporting/IUniverse.sol';
+import 'reporting/IAuctionToken.sol';
+import 'libraries/DelegationTarget.sol';
+import 'libraries/token/VariableSupplyToken.sol';
+import 'libraries/ITyped.sol';
+import 'trading/ICash.sol';
+import 'libraries/Initializable.sol';
+import 'reporting/IAuction.sol';
+import 'reporting/IReputationToken.sol';
+import 'trading/ICash.sol';
+
+
+contract AuctionToken is DelegationTarget, ITyped, Initializable, VariableSupplyToken, IAuctionToken {
+
+    string constant public name = "Auction Token";
+    uint8 constant public decimals = 18;
+    string constant public symbol = "AUC";
+
+    IAuction public auction;
+    IUniverse public universe;
+    ICash public cash;
+    ERC20 public redemptionToken; // The token being auctioned off and recieved at redemption
+    uint256 public auctionIndex;
+
+    function initialize(IAuction _auction, ERC20 _redemptionToken, uint256 _auctionIndex) public beforeInitialized returns(bool) {
+        endInitialization();
+        auction = _auction;
+        universe = auction.getUniverse();
+        cash = ICash(controller.lookup("Cash"));
+        redemptionToken = _redemptionToken;
+        auctionIndex = _auctionIndex;
+        return true;
+    }
+
+    function mintForPurchaser(address _purchaser, uint256 _amount) public returns (bool) {
+        require(msg.sender == address(auction));
+        mint(_purchaser, _amount);
+        return true;
+    }
+
+    function redeem() public returns (bool) {
+        require(auction.getAuctionIndexForCurrentTime() > auctionIndex);
+        uint256 _ownerBalance = balances[msg.sender];
+        uint256 _tokenBalance = redemptionToken.balanceOf(this);
+        uint256 _redemptionAmount = _ownerBalance.mul(_tokenBalance).div(totalSupply());
+        if (redemptionToken == ERC20(cash)) {
+            cash.withdrawEtherTo(msg.sender, _redemptionAmount);
+        } else {
+            redemptionToken.transfer(msg.sender, _redemptionAmount);
+        }
+        burn(msg.sender, _ownerBalance);
```

Comment:
I know withdrawEtherTo is not reentrant, but it feels safer to burn first since we are performing a send?